ivy_tests/test_ivy/test_functional/test_nn/test_losses.py

```diff
@@ -131,16 +131,18 @@ def test_binary_cross_entropy(
         instance_method,
         fw,
         "binary_cross_entropy",
-        true=np.asarray(true[:length], dtype=true_dtype),
-        pred=np.asarray(pred[:length], dtype=pred_dtype),
+        true=np.asarray(true, dtype=true_dtype),
+        pred=np.asarray(pred, dtype=pred_dtype),
         epsilon=epsilon,
     )
 
 
 # sparse_cross_entropy
 @given(
     dtype_and_true=helpers.dtype_and_values(
-        ivy_np.valid_int_dtypes,
+        tuple(
+            set(ivy_np.valid_int_dtypes).intersection(set({"uint8", "int32", "int64"}))
```

Comment:
same as above, this function supports all other integer types in some capacity, and none of these are supported by any of the backends?